Flagsmith.FlagsmithClient/FlagsmithConfiguration.cs

```diff
@@ -1,59 +1,90 @@
-﻿using Microsoft.Extensions.Logging;
+﻿using Flagsmith.Interfaces;
 using System;
 using System.Collections.Generic;
+
 namespace Flagsmith
 {
-    public class FlagsmithConfiguration
+    public class FlagsmithConfiguration : IFlagsmithClientConfig
     {
+        public const string DefaultApiUrl = "https://edge.api.flagsmith.com/api/v1/";
+
         public FlagsmithConfiguration()
         {
-            ApiUrl = "https://edge.api.flagsmith.com/api/v1/";
+            ApiUrl = DefaultApiUrl;
             EnvironmentKey = string.Empty;
             EnvironmentRefreshIntervalSeconds = 60;
         }
+
         /// <summary>
         /// Override the URL of the Flagsmith API to communicate with.
         /// </summary>
         public string ApiUrl { get; set; }
+
         /// <summary>
         /// The environment key obtained from Flagsmith interface.
         /// </summary>
         public string EnvironmentKey { get; set; }
+
         /// <summary>
         /// Enables local evaluation of flags.
         /// </summary>
         public bool EnableClientSideEvaluation { get; set; }
+
         /// <summary>
         /// If using local evaluation, specify the interval period between refreshes of local environment data.
         /// </summary>
         public int EnvironmentRefreshIntervalSeconds { get; set; }
+
         /// <summary>
         /// Callable which will be used in the case where flags cannot be retrieved from the API or a non existent feature is requested.
         /// </summary>
-        public Func<string, Flag> DefaultFlagHandler { get; set; }
-        /// <summary>
-        /// Provide logger for logging polling info & errors which is only applicable when client side evalution is enabled and analytics errors.
-        /// </summary>
-        public ILogger Logger { get; set; }
+        public Func<string, IFlag> DefaultFlagHandler { get; set; }
+
         /// <summary>
         /// if enabled, sends additional requests to the Flagsmith API to power flag analytics charts.
         /// </summary>
         public bool EnableAnalytics { get; set; }
+
         /// <summary>
         /// Number of seconds to wait for a request to complete before terminating the request
         /// </summary>
-        public Double? RequestTimeout { get; set; }
+        public double? RequestTimeout { get; set; }
+
         /// <summary>
         /// Total http retries for every failing request before throwing the final error.
         /// </summary>
         public int? Retries { get; set; }
+
         /// <summary>
         /// Additional headers to add to requests made to the Flagsmith API
         /// </summary>
         public Dictionary<string, string> CustomHeaders { get; set; }
-        public bool IsValid()
+
+        IReadOnlyDictionary<string, string> IFlagsmithClientConfig.CustomHeaders => CustomHeaders;
+
+        public static FlagsmithConfiguration From(
```

Comment:
What's the logic behind having this `From` method rather than a standard constructor?

Reply:
yes can be constructor as well, can change it

Reply:
Unless there's a reason to do it this way then I think it makes more sense to use a constructor. That seems to be more universally accepted (since we're trying to maintain a consistent interface across our SDKs as much as possible).